packages/forklift-console-plugin/src/modules/Plans/utils/helpers/getPlanSummaryStatus.ts

```diff
@@ -0,0 +1,74 @@
+import { PlanData, PlanSummaryStatus } from '../types';
+import {
+  PlanConditionCategory,
+  PlanConditionStatus,
+  PlanConditionType,
+} from '../types/PlanCondition';
+
+export const getPlanSummaryStatus = (data: PlanData): PlanSummaryStatus => {
+  const plan = data?.obj;
+  const conditionTypes = plan?.status?.conditions?.reduce((acc, condition) => {
+    if (condition.status === PlanConditionStatus.True) {
+      acc.push(condition.type);
+    }
+
+    return acc;
+  }, []);
```

Comment:
u are runnindg this array many times using includes,

maybe let's try changing it to obj and then get o(1) extracting the data?
something like
```
const conditionTypes = plan?.status?.conditions?.reduce((acc, condition) => {
    if (condition.status === PlanConditionStatus.True) {
acc[condion.type] = true
    }

    return acc;
  }, {});
```

Reply:
Good idea!